prov/verbs/src/ep_rdm/verbs_tagged_ep_rdm_states.c

```diff
@@ -283,10 +283,10 @@ fi_ibv_rdm_tagged_rndv_rts_send_ready(struct fi_ibv_rdm_tagged_request *request,
 	header->base.tag = request->minfo.tag;
 	header->base.service_tag = 0;
 	header->total_len = request->len;
-	header->src_addr = (uint64_t) (uintptr_t) request->src_addr;
+	header->src_addr = (uintptr_t)request->src_addr;
 
-	header->id = request;
-	request->rndv.id = request;
+	header->id = (uintptr_t)request;
+	request->rndv.id = (uintptr_t)request;
```

Comment:
> > +	header->id = (uintptr_t)request;
> > +	request->rndv.id = (uintptr_t)request;
> 
> Shouldn't this be (uint64_t) request ? Ditto for other casts above and
> below.

If request is a pointer, the cast needs to be uintptr_t, or there will be errors on 32-bit builds.